opentelemetry-etw-metrics/src/exporter/mod.rs

```diff
@@ -69,30 +72,34 @@ fn emit_export_metric_service_request(
 }
 
 impl PushMetricExporter for MetricsExporter {
-    async fn export(&self, metrics: &mut ResourceMetrics) -> OTelSdkResult {
+    async fn export(&self, metrics: &ResourceMetrics) -> OTelSdkResult {
         let schema_url: String = metrics
-            .resource
+            .resource()
             .schema_url()
             .map(Into::into)
             .unwrap_or_default();
 
-        // TODO: Reuse this vec across exports by storing it in `MetricsExporter`
-        let mut encoding_buffer = Vec::new();
+        let resource: Resource = metrics.resource().into();
+        let mut encoding_buffer = Vec::<u8>::with_capacity(etw::MAX_EVENT_SIZE);
```

Comment:
given we serialize one datapoint at a time, pre-allocating the etw_max would be quite excess. Something we can revisit and improve in the future. (Since vec is dropped after the export, not a big issue)